radio/src/thirdparty/Lua/src/lrotable.c

```diff
@@ -10,77 +10,67 @@
 #define LUAR_FINDVALUE        1
 
 /* Utility function: find a key in a given table (of functions or constants) */
-static luaR_result luaR_findkey(const void *where, const char *key, int type, int *found) {
+static luaR_result luaR_findkey(const void * where, const char * key, int type, TValue * found) {
   const char *pname;
   const luaL_Reg *pf = (luaL_Reg*)where;
   const luaR_value_entry *pv = (luaR_value_entry*)where;
   int isfunction = type == LUAR_FINDFUNCTION;
-  *found = 0;
   if(!where)
     return 0;
   while(1) {
     if (!(pname = isfunction ? pf->name : pv->name))
       break;
     if (!strcmp(pname, key)) {
-      *found = 1;
-      return isfunction ? (luaR_result)(size_t)pf->func : (luaR_result)pv->value;
+      if (isfunction) {
+        setlfvalue(found, pf->func);
+      }
+      else {
+        setnvalue(found, pv->value);
+      }
+      return 1;
     }
     pf ++; pv ++;
   }
   return 0;
 }
 
 /* Find a global "read only table" in the constant lua_rotable array */
-luaR_result luaR_findglobal(const char *name, lu_byte *ptype) {
+luaR_result luaR_findglobal(const char * name, TValue * val) {
   unsigned i;
-  *ptype = LUA_TNIL;
-  if (strlen(name) > LUA_MAX_ROTABLE_NAME)
+  if (strlen(name) > LUA_MAX_ROTABLE_NAME) {
+    TRACE_LUA_INTERNALS("luaR_findglobal() '%s' = NAME TOO LONG", name);
     return 0;
+  }
   for (i=0; lua_rotable[i].name; i++) {
     if (!strcmp(lua_rotable[i].name, name)) {
-      *ptype = LUA_TROTABLE;
-      return i+1;
+      setrvalue(val, (void *)(size_t)(i+1));
```

Comment:
This still uses ugly cast, but does not involve double of float. Not sure if we can improve it.